ale_linters/cuda/nvcc.vim

```diff
@@ -0,0 +1,61 @@
+" Author: blahgeek <i@blahgeek.com>
+" Description: NVCC linter for cuda files
+
+call ale#Set('cuda_nvcc_executable', 'nvcc')
+call ale#Set('cuda_nvcc_options', '-std=c++11')
+
+function! ale_linters#cuda#nvcc#GetExecutable(buffer) abort
+    return ale#Var(a:buffer, 'cuda_nvcc_executable')
+endfunction
+
+function! ale_linters#cuda#nvcc#GetCommand(buffer) abort
+    let l:output_file = tempname() . ".ii"
+    call ale#engine#ManageFile(a:buffer, l:output_file)
+    return ale#Escape(ale_linters#cuda#nvcc#GetExecutable(a:buffer))
```

Comment:
I recommend putting a blank line before `return`, `if`, `while`, or `for`, unless they start a new level of indentation.